src/ledctl/ledctl.c

```diff
@@ -694,62 +628,123 @@ struct slot_property *find_slot(struct slot_request *slot_req)
+	if (slot_req->chosen_opt == OPT_SET_SLOT && slot_req->state == LED_IBPI_PATTERN_UNKNOWN) {
 		log_error("Invalid IBPI state in the request.");
 		return LEDCTL_STATUS_INVALID_STATE;
 	}
 	if (slot_req->device[0] && slot_req->slot[0]) {
 		log_error("Device and slot parameters are exclusive.");
 		return LEDCTL_STATUS_DATA_ERROR;
 	}
-	if (slot_req->chosen_opt != OPT_LIST_SLOTS && find_slot(slot_req) == NULL) {
-		log_error("Slot was not found for provided parameters.");
-		return LEDCTL_STATUS_CMDLINE_ERROR;
-	}
+	if (slot_req->chosen_opt != OPT_LIST_SLOTS) {
+		struct led_slot_list_entry *slot = find_slot(ctx, slot_req);
 
+		if (!slot) {
+			log_error("Slot was not found for provided parameters.");
+			return LEDCTL_STATUS_CMDLINE_ERROR;
+		}
+		led_slot_list_entry_free(slot);
+	}
 	return LEDCTL_STATUS_SUCCESS;
 }
 
+
+static inline void print_slot(struct led_slot_list_entry *s)
+{
+	char buf[IPBI2STR_BUFF_SIZE];
+	const char *device_name = led_slot_device(s);
+
+	printf("slot: %-15s led state: %-15s device: %-15s\n",
+		basename(led_slot_id(s)), ibpi2str(led_slot_state(s), buf, sizeof(buf)),
+		(device_name != NULL) ? device_name : "(empty)");
+}
+
+
+/**
+ * @brief List slots connected to given controller
+ *
+ * This function scans all available slots connected to given controller
+ * and prints their led states and names of the connected devices (if exist).
+ *
+ * @param[in]       slot_req       Structure with slot request.
+ *
+ * @return LEDCTL_STATUS_SUCCESS if successful, otherwise ledctl_status_code_t.
+ */
+
+static ledctl_status_code_t list_slots(enum led_cntrl_type cntrl_type)
+{
+	struct led_slot_list_entry *slot = NULL;
+	led_status_t  status = LED_STATUS_SUCCESS;
+	struct led_slot_list *slot_list = NULL;
+
+	status = led_slots_get(ctx, &slot_list);
+	if (status != LED_STATUS_SUCCESS) {
+		log_error("Failed to retrieve slots, reason %s\n",
+				ledctl_strstatus((ledctl_status_code_t)status));
+		return status;
+	}
+
+	while ((slot = led_slot_next(slot_list))) {
+		if (cntrl_type == led_slot_cntrl(slot))
+			print_slot(slot);
+	}
+
+	led_slot_list_free(slot_list);
+	return (ledctl_status_code_t)status;
+}
+
 /**
  * @brief Executes proper slot mode function.
  *
  * @param[in]       slot_req       Structure with slot request.
  *
  * @return LEDCTL_STATUS_SUCCESS if successful, otherwise ledctl_status_code_t.
  */
-ledctl_status_code_t slot_execute(struct slot_request *slot_req)
+ledctl_status_code_t slot_execute(struct led_ctx *ctx, struct slot_request *slot_req)
 {
-	struct slot_property *slot;
+	struct led_slot_list_entry *slot;
 
 	if (slot_req->chosen_opt == OPT_LIST_SLOTS)
 		return list_slots(slot_req->cntrl);
 
-	slot = find_slot(slot_req);
+	slot = find_slot(ctx, slot_req);
 	if (slot == NULL)
 		return LEDCTL_STATUS_DATA_ERROR;
 
 	switch (slot_req->chosen_opt) {
 	case OPT_SET_SLOT:
-		if (get_slot_pattern(slot) == slot_req->state) {
+		led_status_t set_rc = LEDCTL_STATUS_SUCCESS;
```

Comment:
some versions of compilers treat a declaration in 'case' without block statement as an error

`ledctl.c:722:3: error: a label can only be part of a statement and a declaration is not a statement`

pleas add braces around `case OPT_SET_SLOT: block`